packages/react-native-gradle-plugin/build.gradle.kts

```diff
@@ -50,6 +52,19 @@ java {
     targetCompatibility = JavaVersion.VERSION_11
 }
 
+val jvmVersion = Jvm.current()?.javaVersion?.majorVersion
+if (jvmVersion != "11") {
+    println("\n\n\n")
+    println("**************************************************************************************************************")
+    println("\n\n")
+    println("ERROR: react-native-gradle-plugin builds with JVM version 11.")
+    println("Incompatible major version detected: '" + jvmVersion + "'")
+    println("\n\n")
+    println("**************************************************************************************************************")
+    println("\n\n\n")
+    exitProcess(1)
+}
```

Comment:
Thanks for your PR @nomi9995 

Sadly, This won't work for all the cases. I suggest to move this same code to both:
`react.gradle` and the `ReactPlugin.kt`.

The problem is also not that `react-native-gradle-plugin builds with JVM version 11` (that can be addressed by changing the line 51/52 of this same file.

The issue is with AGP 7.x having Java 11 as a hard requirement.

Reply:
@cortinico Thanks for explaining to me. can you check now?

Reply:
@cortinico i don't know why it is failing `ci/circleci: test_js_prev_lts` test. it is working on my local but here is giving an error. can you rerun this test case?

Reply:
Just to note - agp7.x doesn't have java11 as a hard requirement in my experience. It's android API 31+ needing JDK11, you suffer internal compiler errors on jdk8 with compileSdk 31. Same result though, I'm splitting hairs and who knows, might be wrong about AGP7+ just that I did not have problems with jdk8 until I bumped to SDK 31, and was definitely using AGP7